File: router/src/main/java/com/therouter/router/Autowired.java
package com.therouter.router;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Annotation for parameters, which need autowired.
 */
@Retention(RetentionPolicy.SOURCE)
@Target({ElementType.FIELD})
public @interface Autowired {

    String name() default "";

    String args() default "";

    int id() default 0;

    // If required, app will be crash when value is null.
    // Primitive type wont be check!
    boolean required() default false;

    // Description of the field
    String description() default "No desc.";
}

File: agp8-project/apt/src/main/java/com/therouter/router/Autowired.kt
package com.therouter.router

import java.lang.annotation.Retention
import java.lang.annotation.RetentionPolicy

/**
 * Annotation for parameters, which need autowired.
 */
@Retention(RetentionPolicy.SOURCE)
@Target(AnnotationTarget.FIELD)
annotation class Autowired(
    val name: String = "",
    val args: String = "",
    val id: Int = 0,  // If required, app will be crash when value is null.
    // Primitive type wont be check!
    val required: Boolean = false,  // Description of the field
    val description: String = "No desc."
)